state/options/tx_options.go

```diff
@@ -0,0 +1,201 @@
+package options
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+
+const (
+	// gasMultiplier is used to increase gas limit in case if tx has additional options.
+	gasMultiplier = 1.1
+
+	//	Since 0 is a valid fee input for the Tx, the default value is -1.
+	defaultFeeAmount = -1
+)
+
+var (
+	errNoGasProvided     = errors.New("gas limit was not set")
+	errNoAddressProvided = errors.New("address was not set")
+)
+
+// TxOptions specifies additional options that will be applied to the Tx.
+type TxOptions struct {
+	// fee is private since it has to be set through `SetFeeAmount`
+	fee      *Fee
+	GasLimit uint64
```

Comment:
I've always had a problem with `GasLimit` because `Limit` implies the maximum that you can consume (in gas) whereas the Cosmos SDK will always consume that amount, regardless for how much you actually use. It may be better to simply call it `Gas`.

Would a `Gas` of zero indicate that the user wants the gas to be calculated for them?

Reply:
> Would a Gas of zero indicate that the user wants the gas to be calculated for them?

Yes.

>  It may be better to simply call it Gas.

Agree